session.go

```diff
@@ -270,6 +270,13 @@ func (s *Session) init() error {
 		s.hasAggregatesAndFunctions = version.AtLeast(2, 2, 0)
 	}
 
+	reconnectionPolicy := s.cfg.ReconnectionPolicy
+	for i := 0; i < reconnectionPolicy.GetMaxRetries(); i++ {
+		if s.pool.Size() > 0 {
+			break
+		}
+		time.Sleep(reconnectionPolicy.GetInterval(i))
+	}
```

Comment:
I'd prefer if this wait wasn't implemented as polling, since this might wait longer (or just a bit shorter) than necessary depending on the reconnection policy (and the steps there might be exponential).
Instead, we could (for example) close a channel the first time a pool is connected or the initial pool filling fails.

Or, we could remove the wait altogether along with the `if s.pool.Size() == 0 {` condition below, but that will break clients that rely on and check `ErrNoConnectionsStarted`.

Reply:
Thanks @martin-sucha for your insights.
My original idea was similar as "Instead, we could (for example) close a channel the first time a pool is connected or the initial pool filling fails.". 
But the following code snip executes not only during session.init() and I don't want add the code which is only related to session.init().
```
// Fill the connection pool
func (pool *hostConnPool) fill() {
...
        // fill only the first connection synchronously
	if startCount == 0 {
		err := pool.connect()
...
```
"Or, we could remove the wait altogether along with the if s.pool.Size() == 0 { condition below, but that will break clients that rely on and check ErrNoConnectionsStarted." Sounds not bad. I am still thinking if there is any other ways.

How about think of the following definition of session.init [the connection pool part]:
Only all of the connections to those literally configured hosts are ready, session.init() suceeds.
session.init wait until the configured host pool.filling == false, returned the NotAllConfiguredHostsConnected error.